ONSdigital/eq-publisher
# Update mutually exclusive checkbox logic.

### What is the context of this PR?
The pattern for mutually exclusive checkbox answers has recently been updated.

Previously there used to be a specific answer type for mutually exclsive checkboxes. Now, the mutually exclusive pattern can theroetically apply to any answer type by marking the question as being a MutuallyExclusive question type and ensuring that the last answer on the page is the mutually exclusive checkbox option.

This change changes existing logic for checkboxes. After applying this fix, published schema should provide respondents with a separate checkbox answer on the question page as a mutually exclusive option.

### How to review

- Needs code review and all tests and automated checks should pass.

Also worth doing a manual test as a sanity check:
- Checkout master
- Create a questionnaire with a checkbox answer
- Add a mutually exclusive OR option
- Hit launch
- Questionnaire should publish but mutually exclusive option should appear in line with all other regular options.
- Checkout this branch.
- Hit launch again.
- Questionnaire should preview with a separate OR option.
- There should be no validation errors reported.
